### PR TITLE
Implement o-brand.

### PR DESCRIPTION
**Pull request by notlee**
## Overview

- Deprecates mixin `oButtonsCustomTheme`.
- Deprecates color usecases.
- Adds function `oButtonsGetColor`.
- Fixes an existing issue with custom button focus states.

### Why deprecate `oButtonsCustomTheme`?

`oButtons` and `oButtonsTheme` already accept a custom theme map (they delegated to `oButtonsCustomTheme`). `oButtonsCustomTheme` therefore is redundant.

### Why deprecate usecases?

o-brand takes the role of configuring button themes that was previously handled by `o-colors` uses-cases. As `o-brand` should not be used directly outside the origami component this config can be considered private. Unlike `o-colors` usecases it is open to change.

### Why add `oButtonsGetColor`?

It is sometimes useful to match a button color (e.g. [o-tabs does](https://github.com/Financial-Times/o-tabs/blob/master/src/scss/buttontabs.scss#L44) ) which is not possible with deprecated `o-colors` usecases. `oButtonsGetColor` enables `o-buttons` to share its configuration with a getter. 

### Fixes an existing issue with custom button focus states.

Focus states are now set for custom buttons. This is how it was before:

kapture 2018-03-13 at 16 15 50

### CODE REVIEW

**Comment by notlee** [comment]
For the time being button styling remains the same for the internal brand -- except the b2c variant is removed.

File: README.md [hunk 1]
```diff
@@ -182,25 +182,25 @@ Or, using Sass:
 
 #### Theme modifiers
 
-There are 5 named themes in oButtons. oButtons also allows developers to create custom themes using o-colors names.
-
-[View demos](http://registry.origami.ft.com/components/o-buttons)
+There are named themes in oButtons. oButtons also allows developers to create custom themes using o-colors names.
```
Comment (gvonkoss) [inline README.md]: 100% pedantic, but if we aren't refering to the actual `oButtons` JS class, then we should use `o-buttons`?

Comment (gvonkoss) [reply README.md]: 🤔 it's kind of both here, isn't it?

File: main.scss [hunk 2]
```diff
@@ -23,7 +26,7 @@
 		}
 	}
 
-	@each $theme, $properties in $o-buttons-themes {
+	@each $key, $theme in $o-buttons-themes {
```
Comment (gvonkoss) [inline main.scss]: just out of curiosity, do we not use `$key` anywhere?

File: scss/_colors.scss [hunk 3]
```diff
@@ -0,0 +1,6 @@
+@include oColorsSetColor('white-slate-20', oColorsMix($color: 'white', $background: 'slate', $percentage: 20));
+@include oColorsSetColor('slate-white-40', oColorsMix($color: 'slate', $background: 'white', $percentage: 40));
```
Comment (gvonkoss) [inline scss/_colors.scss]: Do any of these colours align with what you've added to `o-colors`?

File: scss/_deprecated.scss [hunk 4]
```diff
@@ -1,3 +1,31 @@
+/// Button custom theme construct
+/// Outputs styles for all button states for custom buttons based on a background color and an accent color
+///
+/// @example
+/// .my-button--custom {
+/// 	@include oButtonsTheme(black, paper);
```
Comment (gvonkoss) [inline scss/_deprecated.scss]: Is this example meant for `oButtonsTheme` or `oButtonsCustomTheme`?

File: scss/_deprecated.scss [hunk 5]
```diff
@@ -65,3 +89,156 @@
...
+
+
+@include oColorsSetColor('white-black-20', oColorsMix($color: 'white', $background: 'black', $percentage: 20));
+
+// Theme: primary
+
+@include oColorsSetUseCase(o-buttons-primary-normal,   text,              'white');
+@include oColorsSetUseCase(o-buttons-primary-normal,   border,            'teal-50');
+@include oColorsSetUseCase(o-buttons-primary-normal,   background,        'teal-50');
+@include oColorsSetUseCase(o-buttons-primary-normal,   _deprecated,       'Use `oButtonsGetColor` instead. Please contact Origami should you have any questions.');
```
Comment (gvonkoss) [inline scss/_deprecated.scss]: This deprecation message won't come up if someone uses a specific property of a usecase, will it?

File: main.scss (hunk 2 above)
Comment (notlee) [reply main.scss]: Guess not! I should be able to remove it

File: scss/_colors.scss (hunk 3 above)
Comment (notlee) [reply scss/_colors.scss]: No, 'slate-white-15` and `slate-white-5` are the new "internal" shades of slate. I left existing slate/white mixes unchanged for now.

File: scss/_deprecated.scss (hunk 4 above)
Comment (notlee) [reply scss/_deprecated.scss]: Good spot. Either way it's wrong. I'll update to `oButtonsCustomTheme`.

File: scss/_deprecated.scss (hunk 5 above)
Comment (notlee) [reply scss/_deprecated.scss]: Will confirm 👍

File: main.scss (hunk 2 above)
Comment (notlee) [reply main.scss]: Actually looks like this is needed in sass to get key/values. http://sass-lang.com/documentation/file.SASS_REFERENCE.html#each-directive

File: scss/_deprecated.scss (hunk 5 above)
Comment (notlee) [reply scss/_deprecated.scss]: It does:

`color: oColorsGetColorFor('o-buttons-primary-normal', 'text');`
screen shot 2018-03-20 at 11 49 34


`@include oColorsFor('o-buttons-primary-normal');`
screen shot 2018-03-20 at 11 51 23

**Comment by notlee** [comment]
Done! As discussed I have also added whitelabel brand config. It provides primary and secondary variants.

kapture 2018-03-20 at 12 40 11

**Pull request merged**